### ISSUE TITLE
Add way to update column names in column transform

### ISSUE THREAD

**Issue by qoega** [ClickHouse/ClickHouse#45618]
**Describe the issue**
We have really nice feature added in #14233 that allows to transform columns and apply function to lists of columns. There is no way to assign column names to resulting columns after APPLY. 
Resulting column names are really unusable when you have if with complex conditions etc.

I would like to have second optional parameter for APPLY with function that applies to column name. Examples:
``` sql
SELECT * APPLY(col -> sum(column), name -> 'sum' || name ) FROM table;
SELECT * APPLY(sum, toLower) FROM table; 
```

**Comment by RachM** [comment]
Any update on this? It really limits the usability of APPLY and REPLACE.

**Comment by mrexfordgala** [comment]
YES! I just found the apply, and it's so nice for adding formatting to strings, but it sadly renames everything to absolute trash. I already named my columns, I'm only applying bulk formatting for readability I can't find a way to rename all the terrible names that are applied. I straight up don't want you to rename it and wish there was a flag to not rename things.




I found an absolutely terrible hack to get around this


```
with prefinal as (select <things named the way I want> )

select * from prefinal where 1=0  -- grab just the column names and data types 
union all
select * Apply i -> f(i)  from prefinal --grab the properly formatted data
```
